include/seqan3/io/sequence_file/format_fasta.hpp

```diff
@@ -329,47 +326,39 @@ class format_fasta
         if (options.fasta_blank_before_id)
             stream_it = ' ';
 
-        std::ranges::copy(id, stream_it);
+        stream_it.write_range(id);
 
         detail::write_eol(stream_it, options.add_carriage_return);
     }
 
     //!\brief Implementation of writing the sequence.
-    template <typename stream_it_t,
-              typename seq_type>
-    void write_seq(stream_it_t & stream_it,
-                   sequence_file_output_options const & options,
-                   seq_type && seq)
+    template <typename stream_it_t, typename seq_type>
+    void write_seq(stream_it_t & stream_it, sequence_file_output_options const & options, seq_type && seq)
     {
+        using distance_type = decltype(std::ranges::distance(seq));
+
+        auto v = seq | views::to_char;
+
         if (options.fasta_letters_per_line > 0)
         {
-        #if SEQAN3_WORKAROUND_VIEW_PERFORMANCE
-            size_t count = 0;
-            for (auto c : seq)
+            auto it = v.begin();
+            distance_type num_lines = std::ranges::distance(seq) / options.fasta_letters_per_line;
```

Comment:
Calling distance might be really expensive if the range is not sized! Fasta files can hold genomes...

Reply:
Well but without knowing the size, I cannot efficiently use the buffer.. I could differentiate between both cases..

Reply:
I removed the dependance on std::distance